crates/matrix-sdk/src/room_preview.rs

```diff
@@ -254,101 +250,181 @@ impl RoomPreview {
     }
 }
 
-pub enum RoomPreviewActions {
-    Invited { join: JoinRoomPreviewAction, leave: LeaveRoomPreviewAction },
-    Knocked { leave: LeaveRoomPreviewAction },
+/// A set of actions that can be performed in a [`RoomPreview`].
+#[derive(Debug)]
+pub struct RoomPreviewActions {
+    /// Action used to join a room.
+    pub join: JoinRoomAction,
+    /// Action used to leave a room.
+    pub leave: LeaveRoomAction,
+    /// Action used to knock on a room.
+    pub knock: KnockRoomAction,
 }
 
-pub struct JoinRoomPreviewAction {
+/// A helper to join a room and perform any needed post-processing if it
+/// succeeds.
+#[derive(Debug)]
+pub struct JoinRoomAction {
     client: Client,
+    room_id: OwnedRoomId,
+    room_state: Option<RoomState>,
+    is_direct: bool,
 }
 
-impl JoinRoomPreviewAction {
-    pub fn new(client: Client) -> Self {
-        Self { client }
+impl RoomPreviewActions {
+    /// Create a new instance given a [`Client`] to perform the actions and a
+    /// [`RoomPreview`] to get the needed data.
+    pub fn new(client: Client, room_preview: RoomPreview) -> Self {
+        let room_id = room_preview.room_id;
+        let room_state = room_preview.state;
+        let is_direct = room_preview.is_direct.unwrap_or_default();
```

Comment:
That's a lie, let's keep an `Option<bool>` for `is_direct` ;)

Reply:
This code was removed.